### ISSUE TITLE
Crash when running with `RUST_BACKTRACE=1`

### ISSUE THREAD

**Issue by p-e-w** [EricLBuehler/mistral.rs#689]
## Describe the bug

```
cargo run -- -i plain -m microsoft_Phi-3-mini-4k-instruct -a phi3
```

No problem, model is loaded and reaching an input prompt.

```
RUST_BACKTRACE=1 cargo run -- -i plain -m microsoft_Phi-3-mini-4k-instruct -a phi3
```

Crashes immediately with

```
2024-08-16T12:25:08.459036Z  INFO mistralrs_core::pipeline::normal: Loading `tokenizer_config.json` locally at `microsoft_Phi-3-mini-4k-instruct/tokenizer_config.json`
Error: unsupported dtype BF16 for op matmul
   0: candle_core::error::Error::bt
             at ~/.cargo/git/checkouts/candle-c6a149c3b35a488f/9e09d7f/candle-core/src/error.rs:231:25
   1: <candle_core::cpu_backend::MatMul as candle_core::cpu_backend::utils::Map2>::f
             at ~/.cargo/git/checkouts/candle-c6a149c3b35a488f/9e09d7f/candle-core/src/cpu_backend/mod.rs:1260:22
   2: candle_core::cpu_backend::utils::Map2::map
             at ~/.cargo/git/checkouts/candle-c6a149c3b35a488f/9e09d7f/candle-core/src/cpu_backend/utils.rs:47:54
   3: <candle_core::cpu_backend::CpuStorage as candle_core::backend::BackendStorage>::matmul
             at ~/.cargo/git/checkouts/candle-c6a149c3b35a488f/9e09d7f/candle-core/src/cpu_backend/mod.rs:2433:9
   4: candle_core::storage::Storage::matmul
             at ~/.cargo/git/checkouts/candle-c6a149c3b35a488f/9e09d7f/candle-core/src/storage.rs:719:31
   5: candle_core::tensor::Tensor::matmul
             at ~/.cargo/git/checkouts/candle-c6a149c3b35a488f/9e09d7f/candle-core/src/tensor.rs:1258:23
   6: mistralrs_core::utils::normal::determine_auto_dtype
             at ./mistralrs-core/src/utils/normal.rs:129:17
   7: <mistralrs_core::utils::normal::ModelDType as mistralrs_core::utils::normal::TryIntoDType>::try_into_dtype
             at ./mistralrs-core/src/utils/normal.rs:183:30
   8: <mistralrs_core::pipeline::normal::NormalLoader as mistralrs_core::pipeline::Loader>::load_model_from_path
             at ./mistralrs-core/src/pipeline/normal.rs:231:21
   9: <mistralrs_core::pipeline::normal::NormalLoader as mistralrs_core::pipeline::Loader>::load_model_from_hf
             at ./mistralrs-core/src/pipeline/normal.rs:208:9
  10: mistralrs_server::main::{{closure}}
             at ./mistralrs-server/src/main.rs:412:20
  11: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/future/future.rs:123:9
  12: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/park.rs:281:63
  13: tokio::runtime::coop::with_budget
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/coop.rs:107:5
  14: tokio::runtime::coop::budget
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/coop.rs:73:5
  15: tokio::runtime::park::CachedParkThread::block_on
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/park.rs:281:31
  16: tokio::runtime::context::blocking::BlockingRegionGuard::block_on
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/context/blocking.rs:66:9
  17: tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/scheduler/multi_thread/mod.rs:87:22
  18: tokio::runtime::context::runtime::enter_runtime
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/context/runtime.rs:65:16
  19: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/scheduler/multi_thread/mod.rs:86:9
  20: tokio::runtime::runtime::Runtime::block_on_inner
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/runtime.rs:363:50
  21: tokio::runtime::runtime::Runtime::block_on
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/runtime.rs:333:13
  22: mistralrs_server::main
             at ./mistralrs-server/src/main.rs:478:5
  23: core::ops::function::FnOnce::call_once
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/ops/function.rs:250:5
  24: std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:155:18
  25: std::rt::lang_start::{{closure}}
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:159:18
  26: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/ops/function.rs:284:13
  27: std::panicking::try::do_call
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:559:40
  28: std::panicking::try
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:523:19
  29: std::panic::catch_unwind
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panic.rs:149:14
  30: std::rt::lang_start_internal::{{closure}}
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:141:48
  31: std::panicking::try::do_call
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:559:40
  32: std::panicking::try
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:523:19
  33: std::panic::catch_unwind
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panic.rs:149:14
  34: std::rt::lang_start_internal
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:141:20
  35: main
  36: __libc_start_call_main
             at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  37: __libc_start_main_impl
             at ./csu/../csu/libc-start.c:392:3
  38: _start


Stack backtrace:
   0: anyhow::error::<impl anyhow::Error>::msg
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/anyhow-1.0.86/src/backtrace.rs:27:14
   1: core::ops::function::FnOnce::call_once
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/ops/function.rs:250:5
   2: core::result::Result<T,E>::map_err
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/result.rs:854:27
   3: <mistralrs_core::utils::normal::ModelDType as mistralrs_core::utils::normal::TryIntoDType>::try_into_dtype
             at ./mistralrs-core/src/utils/normal.rs:183:30
   4: <mistralrs_core::pipeline::normal::NormalLoader as mistralrs_core::pipeline::Loader>::load_model_from_path
             at ./mistralrs-core/src/pipeline/normal.rs:231:21
   5: <mistralrs_core::pipeline::normal::NormalLoader as mistralrs_core::pipeline::Loader>::load_model_from_hf
             at ./mistralrs-core/src/pipeline/normal.rs:208:9
   6: mistralrs_server::main::{{closure}}
             at ./mistralrs-server/src/main.rs:412:20
   7: <core::pin::Pin<P> as core::future::future::Future>::poll
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/future/future.rs:123:9
   8: tokio::runtime::park::CachedParkThread::block_on::{{closure}}
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/park.rs:281:63
   9: tokio::runtime::coop::with_budget
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/coop.rs:107:5
  10: tokio::runtime::coop::budget
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/coop.rs:73:5
  11: tokio::runtime::park::CachedParkThread::block_on
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/park.rs:281:31
  12: tokio::runtime::context::blocking::BlockingRegionGuard::block_on
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/context/blocking.rs:66:9
  13: tokio::runtime::scheduler::multi_thread::MultiThread::block_on::{{closure}}
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/scheduler/multi_thread/mod.rs:87:22
  14: tokio::runtime::context::runtime::enter_runtime
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/context/runtime.rs:65:16
  15: tokio::runtime::scheduler::multi_thread::MultiThread::block_on
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/scheduler/multi_thread/mod.rs:86:9
  16: tokio::runtime::runtime::Runtime::block_on_inner
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/runtime.rs:363:50
  17: tokio::runtime::runtime::Runtime::block_on
             at ~/.cargo/registry/src/index.crates.io-6f17d22bba15001f/tokio-1.39.1/src/runtime/runtime.rs:333:13
  18: mistralrs_server::main
             at ./mistralrs-server/src/main.rs:478:5
  19: core::ops::function::FnOnce::call_once
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/ops/function.rs:250:5
  20: std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/sys_common/backtrace.rs:155:18
  21: std::rt::lang_start::{{closure}}
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:159:18
  22: core::ops::function::impls::<impl core::ops::function::FnOnce<A> for &F>::call_once
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/core/src/ops/function.rs:284:13
  23: std::panicking::try::do_call
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:559:40
  24: std::panicking::try
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:523:19
  25: std::panic::catch_unwind
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panic.rs:149:14
  26: std::rt::lang_start_internal::{{closure}}
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:141:48
  27: std::panicking::try::do_call
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:559:40
  28: std::panicking::try
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panicking.rs:523:19
  29: std::panic::catch_unwind
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/panic.rs:149:14
  30: std::rt::lang_start_internal
             at /rustc/051478957371ee0084a7c0913941d2a8c4757bb9/library/std/src/rt.rs:141:20
  31: main
  32: __libc_start_call_main
             at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  33: __libc_start_main_impl
             at ./csu/../csu/libc-start.c:392:3
  34: _start
```

I had no idea it is possible for `RUST_BACKTRACE=1` to *cause* a crash. I couldn't find a single report online where that is the case.

Note that this is the same crash reported in #669, except that

* it's a different model
* it only happens with `RUST_BACKTRACE=1`

## Latest commit or version

df5583e87e7e4a9fab1296a8deffe0ec94580b8e

**Comment by EricLBuehler** [comment]
@p-e-w this should be fixed on `master` and `dry_penalty` as I just merged #690.

**Comment by p-e-w** [comment]
@EricLBuehler 

Confirmed fixed.
